### ISSUE TITLE
rolling upgrade for container based deployments

### ISSUE THREAD

**Issue by leseb** [ceph/ceph-ansible#1061]
Address upgrades of containerized deployments.

I think we have two main options:

* run the rolling_update playbook and let the containerized ansible tasks taking care of the upgrade.
If the logic can not be handled with the current rolling update playbook then we should disable https://github.com/ceph/ceph-ansible/blob/master/infrastructure-playbooks/rolling_update.yml (not allow it to run) if we deployed a containerized deployment with it.

* add a new playbook under `infrastructure-playbooks` that is supposed to upgrade containerized ceph clusters. Will be really similar to https://github.com/ceph/ceph-ansible/blob/master/infrastructure-playbooks/rolling_update.yml 

@font do you mind giving this a try?
Thanks!

**Comment by font** [comment]
@leseb Will do.

So upon initial testing I ran into the error:
```
TASK [ceph-common : make sure an installation source was chosen] ***************
fatal: [mon0]: FAILED! => {"changed": false, "failed": true, "msg": "choose an upstream installation source or read https://github.com/ceph/ceph-ansible/wiki"}
```

This is because `mon_containerized_deployment` was not being picked up properly before running the roles. So I am in the process of shuffling things around and seeing if I can get it to work using the same playbook. I'd really like to avoid having another similar playbook just for containerized deployments. Stay tuned...

**Comment by leseb** [comment]
LGTM, yes we could try to do that for purge as well ;)
Thanks @font!

**Comment by font** [comment]
@leseb Haha, yes we could. But I think the purge is a bit more complicated because we're trying to undo what the deployment playbook has done, which is a bit different between containerized and non-containerized deployments. Whereas for the upgrade playbook, we're trying to see if we can get away with just re-running the roles.

**Comment by leseb** [comment]
Addressed in #1077
